src/lib/KeepaliveManager.ts

```diff
@@ -0,0 +1,89 @@
+import type MqttClient from './client'
+import getTimer, { type Timer } from './get-timer'
+import type { TimerVariant } from './shared'
+
+export default class KeepaliveManager {
+	private _keepalive: number
+
+	private timerId: number
+
+	private timer: Timer
+
+	private destroyed = false
+
+	private counter: number
+
+	private client: MqttClient
+
+	private _keepaliveTimeoutTimestamp: number
+
+	/** Timestamp of next keepalive timeout */
+	get keepaliveTimeoutTimestamp() {
+		return this._keepaliveTimeoutTimestamp
+	}
+
+	set keepalive(value: number) {
+		if (
+			// eslint-disable-next-line no-restricted-globals
+			isNaN(value) ||
+			value < 0 ||
+			value * 1000 > 2147483647
```

Comment:
whyh not multiply with 1000 bevore using it ... now you multiply 2 times

Reply:
because I don't want to set it before validating it, the alternative is to write `value = value * 1000` on the above line, I can do that